[Park Seoyoung] iP

Judy1x4 opened the pull request:
# Jude the chatbot
- This is Jude the chatbot your friendly personal task assistant.

It helps you to:
1. Add a task
2. Delete a task
3. View a current task list
4. Find a task from a task list

Moreover, different types of tasks can be added, for example:
```
to-do read book
deadline homework /by today
event meeting /from 8 /to 9
```
- [x] ~~Typing-preferred :+1:~~
- [ ] _Easy to use_
- [ ] Personalised assistence

> [!NOTE]
> This project is under NUS in the course CS2103T.

To get more information about the course, visit [CS2103T webpages](https://nus-cs2103-ay2425s2.github.io/website/).

**to install**:
1. Copy the jar file into an empty folder.
2. Open a command window in that folder.
3. Run the command `java -jar "Jude.jar"` (i.e., run the command in the same folder as the jar file).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Judy1x4 committed:
Edit .gitignore

Judy1x4 committed:
Rename chatbot to Jude

Judy1x4 committed:
Implement an initial skeletal version

Judy1x4 committed:
Store echoing of user input and display upon request

Judy1x4 committed:
Correct Java layout and add header comments

Judy1x4 committed:
Task class: Add Task class

Judy1x4 committed:
Add Javadoc to Task class

Judy1x4 committed:
Add mark and unmark of tasks in Jude class

Judy1x4 committed:
Jude class: Add Javadoc

Judy1x4 committed:
Create subclasses of Task

There are different tasks which can be grouped into new class.

This enables more specific task management.

Create subclasses of Task with different fields.

Let's
* Deadline class: contains information about deadline
* Event class: contains from and to date/time
* Todo: a simple task without deadline or duration

Using inheritance is preferable due to is-a relationship.

Judy1x4 committed:
Jude class: Add parser object to replace slicing inputs

Judy1x4 committed:
Change reading todo input to to-do input: distinguish with todo comment

Judy1x4 committed:
Use i/o redirection technique to semi-automate the regression testing

Judy1x4 committed:
Add JudeException class handling invalid input

Judy1x4 committed:
Replace code with JudeException where necessary

Judy1x4 committed:
Use JudeException in Parser class

Judy1x4 committed:
Add invalid test cases

Judy1x4 committed:
Jude: Correct to show full task details when task added

Judy1x4 committed:
Replace Task array to Task Arraylist. Add remove command

Judy1x4 committed:
Add save feature

Judy1x4 committed:
Merge branch 'branch-Level-7'

Judy1x4 committed:
Work in progress: Replace String date or time in Deadline and Event

Judy1x4 committed:
Complete replacement of String to Time or Date object

Judy1x4 committed:
Work in progress: Combine Date and Time object to DateTime object

Judy1x4 committed:
Complete in Replacing Date and Time object to DateTime object
There are Time and Date objects in Deadline and Event class.

Unifying all those fields improves code quality.

Let's:
* extract those complicated fields into unified DateTime object
* Deadline and Event constructor parameters are simplified
* loadFile split logic is simpler.

Judy1x4 committed:
Ignore jude.txt save file in data directory

Judy1x4 committed:
Merge branch 'branch-Level-8'

Judy1x4 committed:
Add Storage, TaskList, Ui classes and replace main method appropriately

Judy1x4 committed:
Add Command classes

Judy1x4 committed:
Add more methods

Judy1x4 committed:
Fix Command and Jude classes

Let's
* fix delete command to properly delete the index
* fix invalid index input
* fix typos in command classes
* initialize fields in Jude class

Judy1x4 committed:
Fix index logic and typo

Let's
* fix accessing deleted command from list, causing an error
* fix index to be zero-indexed in Delete constructor
* fix minor typo

Judy1x4 committed:
Merge branch 'branch-A-MoreOOP'

Judy1x4 committed:
Update jude package to all class and command, task packages

Judy1x4 committed:
Merge branch 'add-gradle-support'

Judy1x4 committed:
Merge branch 'branch-A-Packages'

Judy1x4 committed:
Override toString in command classes

Judy1x4 committed:
Fix unhandled exception and typo

Judy1x4 committed:
Add Test classes and write test cases in some classes

Judy1x4 committed:
Update Gradle configuration

Judy1x4 committed:
Fix UiTest with correct content

Judy1x4 committed:
Merge branch 'branch-A-JUnit'

Judy1x4 committed:
Write JavaDoc comments for all non-trivial methods and classes

Judy1x4 committed:
Adjust blank line separation and unused field

Judy1x4 committed:
Add find command

Judy1x4 committed:
Merge branch 'branch-Level-9'

Judy1x4 committed:
Merge branch

Judy1x4 committed:
Resolve conflict

jon-seana commented on src/main/java/jude/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+ * to keep track of their pending tasks easily.
+ */
+public class TaskList {
+    private List<Task> list;
```
Should this be renamed to "lists" instead of "list" as it is a multi valued variable.

wei-song1 replied:
Agreed

jon-seana commented on src/main/java/jude/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+        return list.get(index);
+    }
+
+    public int size() {
```
should this method be renamed to getSize(), as method names should be verbs.

wei-song1 replied:
Agreed, coding standard says "Names representing methods must be verbs and written in camelCase."

jon-seana commented on src/main/java/jude/command/AddCommand.java:
```diff
@@ -0,0 +1,37 @@
+ * Represents the class which contains the series of actions to add a Task into the TaskList to be executed.
+ */
+public class AddCommand extends Command {
+    private Task task;
```
Goodjob on putting the correct naming convention. "task" is good as it is a single valued variable.

jon-seana commented on src/main/java/jude/command/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    /**
+     * Deletes task from the TaskList. Notifies the user that the task has been deleted.
+     * @param list that will store the task deleted
```
I believe there should be a space after the description line for the method in JavaDoc comment format.

jon-seana commented on src/main/java/jude/command/MarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+    public MarkCommand(int index) {
+        this.index = index;
+    }
+    @Override
```
Would this be a labelling misplacement?

wei-song1 replied:
Agreed, the `@Override` seems to be in a strange position

jon-seana reviewed (commented):
Looks very neat and good.
Some minor cosmetic changes requested.

wei-song1 commented on src/main/java/jude/Jude.java:
```diff
@@ -0,0 +1,55 @@
+import jude.command.Command;
+
+/**
+ * Represents a jude.Jude, the personal assistant chatbot.
```
Minor mistake in jude.Jude from what I am guessing is renaming the file name and IntelliJ trying to fix it, consider fixing it

wei-song1 commented on src/main/java/jude/Jude.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class Jude {
+
+    String name = "jude.Jude";
```
Same minor mistake as the earlier one in the comment

wei-song1 commented on src/main/java/jude/Storage.java:
```diff
@@ -0,0 +1,119 @@
+     * Loads the save file. Creates the file if the file does not exist in the filePath.
+     * @return TaskList containing the information read from the save file.
+     * @throws JudeException if there was error loading the file.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the `Loads the save file...` and `@return TaskList...`

wei-song1 commented on src/main/java/jude/Storage.java:
```diff
@@ -0,0 +1,119 @@
+        return list;
+    }
+
+    /**
+     * Saves the current Tasklist to the save file.
+     * @param list the TaskList created during the execution of Jude chatbot program.
+     * @throws JudeException if there was an error while creating a save file.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/Ui.java:
```diff
@@ -0,0 +1,53 @@
+        System.out.println("Poyo. Hope to see you again soon!");
+    }
+
+    /**
+     * Read the user input.
+     * @return the user input as String.
+     * @throws JudeException if there is any IOException.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/command/AddCommand.java:
```diff
@@ -0,0 +1,37 @@
+        this.task = task;
+    }
+
+    /**
+     * Adds task to the TaskList. Notifies the user that the task has been added.
+     * @param list that will store the task added.
+     * @param ui displays the message that the task has been added
+     * @param storage will write the task data onto the save file
+     * @throws JudeException, if any one of the method fails
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/command/ExitCommand.java:
```diff
@@ -0,0 +1,29 @@
+ */
+public class ExitCommand extends Command {
+
+    /**
+     * Exit the program. Sets the isExit to true, which is to be tracked in the main method of the chatbot.
+     * @param list
+     * @param ui
+     * @param storage
+     * @throws JudeException, if any one of the method call fails
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/command/ListCommand.java:
```diff
@@ -0,0 +1,29 @@
+ */
+public class ListCommand extends Command {
+
+    /**
+     * List the tasks in the TaskList. Displays the taskList onto the ui.
+     * @param list that will return the string representation of the tasklist
+     * @param ui displays the String representation of the tasklist
+     * @param storage
+     * @throws JudeException, if any one of the method call fails
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/command/MarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+    }
+    @Override
+
+    /**
+     * Marks the task as done. Save the changes into the save file.
+     * @param list
+     * @param ui displays the message that a task has been marked
+     * @param storage will save the marked version of task data to the save file
+     * @throws JudeException, if any one of the method call fails
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+        this.index = index;
+    }
+
+    /**
+     *Unmarks the task as done. Save the changes into the save file.
+     * @param list
+     * @param ui displays the message that a task has been unmarked
+     * @param storage will save the unmarked version of task data to the save file
+     * @throws JudeException, if any one of the method call fails
+     */
```
First line for description appears to be missing a space between the * and text

According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/task/Deadline.java:
```diff
@@ -0,0 +1,67 @@
+public class Deadline extends Task {
+    private LocalDateTime dateTime;
+
+    /**
+     * Creates the Deadline object.
+     * @param description of the Task.
+     * @param deadline of which the format is expected to be of the direct user input.
+     * @throws JudeException, if the format of the deadline is not in the expected format.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/task/Deadline.java:
```diff
@@ -0,0 +1,67 @@
+     * @param dateTime of which the format is expected to be of the save file format.
+     * @param isDone represents that status of the task isDone.
+     * @throws JudeException, if the format of the deadline is not in the expected format.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/task/Deadline.java:
```diff
@@ -0,0 +1,67 @@
+        this.dateTime = LocalDateTime.parse(dateTime);
+    }
+
+    /**
+     * Takes in the String representation of the date and time of a particular format.
+     * Parses the String representation to LocalDateTime object.
+     * @param deadline, the String, which parsed from the user input, containing information about the deadline.
+     * @throws JudeException, if the format of the input is not valid.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+    private LocalDateTime fromDateTime;
+    private LocalDateTime toDateTime;
+
+    /**
+     * Creates the Deadline object.
+     * @param description of the Task.
+     * @param from represents the starting time, which the format is expected to be of the direct user input.
+     * @param to represents the ending time, which the format is expected to be of the direct user input.
+     * @throws JudeException, if the format of the time is not in the expected format.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 commented on src/main/java/jude/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+     * @param toDateTime represents the ending time, which the format is expected to be of the save file format.
+     * @param isDone represents that status of the task isDone.
+     * @throws JudeException, if the format of the time is not in the expected format.
+     */
```
According to the coding standard, "Empty line between description and parameter section", there should be an empty line between the description and the parameters

wei-song1 reviewed (commented):
Beyond very minor mistakes, your code looks amazing, and it looks like you did a really good job in maintaining it.

Judy1x4 committed:
Change formatting in Parser class

Judy1x4 committed:
Fix index error not matching

Judy1x4 committed:
Replace redundant isExit boolean variable and fix typo in Event header

It fixes according to the pull request comments.

Judy1x4 committed:
Add some of worked Javafx GUI

Judy1x4 committed:
Command class: Store task message

Let's
* set the field message.
* get the field message.
* to be passed as an argument for Jude:getResponse.

Judy1x4 committed:
Add profile images

Judy1x4 committed:
Add label according to the Command type

Let's
* get DialogBox with different color for different commandType.

Judy1x4 committed:
Fix typos and add UnmarkCommand back

Judy1x4 committed:
Make parse a static method

Judy1x4 committed:
Make parse a static method

Judy1x4 committed:
Merge branch 'branch-Level-10'

Judy1x4 committed:
Correct some checkstyle error

Judy1x4 committed:
Merge branch 'branch-A-Checkstyle'

Judy1x4 committed:
Add Ui to show message

Judy1x4 committed:
Remove Ui class and related file containing the Ui variable

Judy1x4 committed:
Remove Ui further from the main class

Judy1x4 committed:
Add package ui and add DialogBox and MainWindow classes

Judy1x4 committed:
Fix program not exiting properly

Judy1x4 committed:
Fix style issues

Judy1x4 committed:
Fix small style issue

Judy1x4 committed:
Add assert in Main class that jude cannot be null

There is an assumption that the default file path is correct. In
some cases where the parent directory is not present, the program
will output IOException when loading the save file.

Asserting the path existence assures the correct code behaviour.

As a step towards such correction, let's assert the jude instance
cannot be null.

Judy1x4 committed:
Refactor methods to adhere to SLAP

There are several methods that has a mix of low-level and
high-level implementation at a same scope. In some cases there
are many lines of low-level codes only.

Refactoring all those low-level code or abstracting with comments
improves the code quality.

As a step towards such refactoring, let's:
* extract those low-level code blocks into seperate methods
in their respective classes.
* add high-level comment to pack in two abstraction levels.
Doing so will make the subsequent coding quality improvement
easier.

Judy1x4 committed:
Import binary operator in TaskList class

Judy1x4 committed:
Replace for loop in Tasklist class with stream

Judy1x4 committed:
Merge branch 'branch-A-CodeQuality'

Judy1x4 committed:
Merge branch 'branch-A-Streams'

Judy1x4 committed:
Add Sort Command class

Judy1x4 committed:
Mergt
branch 'branch-BCD-Extension'

Judy1x4 committed:
Correct JavaDoc header to a correct format

Judy1x4 committed:
Debug Junit

Judy1x4 committed:
Fix minor typo

Judy1x4 committed:
Add Ui image

Judy1x4 committed:
Update README.md

Judy1x4 committed:
Update release page

Judy1x4 committed:
Merge branch 'branch-A-UserGuide'

Judy1x4 committed:
Fix gradle build dependency setting

Judy1x4 committed:
Merge pull request #2 from Judy1x4/branch-for-merge

Fix gradle build dependency setting

Judy1x4 committed:
Fix parent directory of data file to be created if not exist

Judy1x4 committed:
Fix checkstyle error

Judy1x4 committed:
Merge branch 'master' of https://github.com/Judy1x4/ip
